src/overlays/actors/ovl_Boss_01/z_boss_01.c

```diff
@@ -246,191 +647,2934 @@ ActorInit Boss_01_InitVars = {
+        Boss01_SetupWait(this, play, ODOLWA_WAIT_RANDOM);
+    }
+
+    this->kickAndShieldBashCollisionEnabled = true;
+}
+
+/**
+ * If Odolwa is not currently in the "damaged" state (where he's crouched on one knee and breathing heavily), this
+ * function will transition him to that state and start a 20 frame timer; when this timer reaches 0, he will recover and
+ * jump away. However, if he is already in the "damaged" state, and if the player attacked him with an attack that has
+ * the ODOLWA_DMGEFF_DAMAGE_TIMER_CHECK damage effect, then one of two things will happen:
+ * - If the attack hit while his current action timer is 7 or more, the timer will be set to 20 again; this will keep
+ *   Odolwa in the "damaged" state for longer and allow the player to attack him more.
+ * - If the attack hit while his current action timer is 6 or less, Odolwa will disable all of his collision for 20
+ *   frames to ensure that the player cannot hit him with subsequent attacks.
+ */
+void Boss01_SetupDamaged(Boss01* this, PlayState* play, u8 damageEffect) {
+    if (this->actionFunc != Boss01_Damaged) {
+        this->timers[TIMER_CURRENT_ACTION] = 20;
+        Animation_MorphToPlayOnce(&this->skelAnime, &gOdolwaDamagedStartAnim, 0.0f);
+        this->animEndFrame = Animation_GetLastFrame(&gOdolwaDamagedStartAnim);
+        this->actionFunc = Boss01_Damaged;
+    } else if (damageEffect == ODOLWA_DMGEFF_DAMAGE_TIMER_CHECK) {
+        if (this->timers[TIMER_CURRENT_ACTION] > 5) {
+            this->disableCollisionTimer = 20;
+        } else {
+            this->timers[TIMER_CURRENT_ACTION] = 20;
+        }
+    }
+}
+
+/**
+ * Plays Odolwa's damaged animation until the current action timer reaches 0, then makes him jump away. If the player
+ * attacks Odolwa while he's "downed" in this state, the current action timer can be manipulated in Boss01_SetupDamaged,
+ * so it's possible for Odolwa to stay in this state for a long time with well-timed attacks.
+ */
+void Boss01_Damaged(Boss01* this, PlayState* play) {
+    SkelAnime_Update(&this->skelAnime);
+    Math_ApproachZeroF(&this->actor.speed, 1.0f, 1.0f);
+
+    if (Animation_OnFrame(&this->skelAnime, this->animEndFrame)) {
+        Animation_MorphToLoop(&this->skelAnime, &gOdolwaDamagedLoopAnim, this->animMorphFrames2);
+        this->animEndFrame = 1000.0f;
+    }
+
+    if (this->timers[TIMER_CURRENT_ACTION] == 0) {
+        Boss01_SetupJump(this, play, false);
+    }
+
+    Boss01_SpawnDustAtFeet(this, play, 1);
+}
+
+void Boss01_UpdateDamage(Boss01* this, PlayState* play) {
+    Player* player = GET_PLAYER(play);
+    u8 damage;
+    s32 i;
+
+    if (this->shieldCollider.elements[ODOLWA_SHIELD_COLLIDER_SHIELD].info.bumperFlags & BUMP_HIT) {
+        this->bodyInvincibilityTimer = 5;
+        if (this->damageTimer == 0) {
+            ColliderInfo* acHitInfo = this->shieldCollider.elements[ODOLWA_SHIELD_COLLIDER_SHIELD].info.acHitInfo;
+
+            if (acHitInfo->toucher.dmgFlags == 0x2000000) {
```

Comment:
```suggestion
            if (acHitInfo->toucher.dmgFlags == DMG_SWORD_BEAM) {
```

Reply:
There's a comment right above this in `z64collision_check.h` that says:
```c
// These flags are not to be used in code until we figure out how we want to format them. They are only here for reference 
```

So I didn't use them in the code. Should I just ignore that comment? Should we start using these everywhere?

Reply:
Let me know if I'm wrong, but I believe single flags are fine. It is combinations of multiple flags that we are waiting on before using them everywhere.

Reply:
Yeah, single flags are fine.
If you look up for those flags in the repo you'll see they are used for single but left untouch for combinations. Sometimes people leave a comment for combinations to make it a bit more readable (`DMG_SWORD_BEAM | DMG_SPIN_ATTACK | ...`)

Reply:
I updated the comment in `z64collision_check.h` to try to make it accurate to this status quo. Let me know what you think.